cmd/autoprowjobdispatcher/main.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/experiment/autobumper/bumper"
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/labels"
+)
+
+const (
+	githubOrg    = "openshift"
+	githubRepo   = "release"
+	githubLogin  = "openshift-bot"
+	githubTeam   = "openshift/openshift-team-developer-productivity-test-platform"
+	matchTitle   = "Automate config brancher"
+	remoteBranch = "auto-config-brancher"
+)
+
+type options struct {
+	selfApprove bool
+
+	githubLogin string
+	gitName     string
+	gitEmail    string
+	targetDir   string
+	assign      string
+
+	prometheusPasswordPath string
+
+	flagutil.GitHubOptions
+}
+
+func parseOptions() options {
+	var o options
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.githubLogin, "github-login", githubLogin, "The GitHub username to use.")
```

Comment:
Instead of having more and more periodic jobs where the result of one might impact the result of others, can we just make this part of the `autoconfigbrancher` and remove all the pr-creation functionality?

Reply:
2 excuses that I did not add this into `auto-config-brancher` process:

- In the card, we plan to run this tool weekly. However, the brancher job is hourly. I do not think it is a big deal (even it is hourly) since the current implementation respects the previous dispatching results. So the next is the real reason.
-  I would start with `self-approve=false` to build up my confidence that the process does the right thing. (I have run the tool locally several times but still I want to be careful).

Reply:
Ok, the different cadence is a good point. I think you can canary out the thing locally, no need to use `self-approve=false`